blockly-ws-search/src/WorkspaceSearch.js

```diff
@@ -409,27 +422,40 @@ export class WorkspaceSearch {
   }
 
   /**
-   * Searches the workspace for the current search term.
-   * @param {Blockly.Block=} opt_block Block to use as the current
+   * Overrides the default behavior of the search function.
+   * @param {?function(Blockly.WorkspaceSvg, string): Array.<Blockly.BlockSvg>} customSearchFunction
+   *    The function to use instead of the default search or null to reset to
+   *    default behavior.
+   */
+  setSearchFunction(customSearchFunction) {
+    this.customSearchFunction_ = customSearchFunction;
+  }
+
+  /**
+   * Searches the workspace for the current search term and highlights found
+   * blocks.
+   * @param {Blockly.BlockSvg=} opt_block Block to use as the current
    *     block if it is included in new search blocks.
    */
-  search(opt_block) {
+  searchAndHighlight(opt_block) {
```

Comment:
Could this be private? Since we now allow them to set the search function, and they can change how the block is highlighted through css?

Reply:
Leaving this public gives the freedom of triggering/binding it outside of the search bar.